### DISCUSSION TITLE
Return type while using pojo

### DISCUSSION THREAD

**Discussion by mastermunj** [adonisjs/core#2595] [Help]
When we use pojo as 
```ts
const count = await SomeModel.query().pojo<{ count: string }>().count('*');
```
Type of count = `{ count: string }[]`

When used with `.first()`
```ts
const count = await SomeModel.query().pojo<{ count: string }>().count('*').first();
```
Type of count = `{ count: string } | null`

Shouldn't it be `{ count: string }` with `.first()`?

**Comment by mastermunj** [comment]
@thetutlage Please help.

**Reply by thetutlage** [reply]
It returns `{ count: string } | null` because the `first` method picks the first item from the array and if the array is empty it returns null.

In your case, the array won't be empty because you are using an aggregate. However, the query builder doesn't switch types when an aggregate is defined and treat all queries as same.

This can be changed, but I am not sure how big the change will be.

**Reply by mastermunj** [reply]
Understood.

For now I am doing something like
```ts
const [{ count }] = await this.query().pojo<{ count: number }>().count('*').where(values);
```
which gets me perfect balance between type check and readability.
